app/controllers/organization_authorized_controller.rb

```diff
@@ -0,0 +1,19 @@
+class OrganizationAuthorizedController < ApplicationController
+  before_action :set_organization, :authorize_organization_access
+
+  decorates_assigned :organization
+
+  private
+
+  def authorize_organization_access
+    not_found if !@organization.users.include?(current_user) && current_user.staff?
```

Comment:
@mastahyeti beat me to it, I had a comment box open for that line and was trying to parse it mentally.